x-pack/plugin/ccr/src/internalClusterTest/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java

```diff
@@ -514,13 +514,53 @@ public void testPauseAndResumeWithMultipleAutoFollowPatterns() throws Exception
         assertThat(followerClient().admin().indices().prepareStats("copy-*").get().getIndices().size(), equalTo(leaderIndices.get()));
     }
 
+    public void testAutoFollowExclusion() throws Exception {
+        Settings leaderIndexSettings = Settings.builder()
+            .put(IndexMetadata.INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), 1)
+            .put(IndexMetadata.INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), 0)
+            .build();
+
+        putAutoFollowPatterns("my-pattern1", new String[] {"logs-*"}, Collections.singletonList("logs-2018*"));
+
+        createLeaderIndex("logs-201701", leaderIndexSettings);
+        assertLongBusy(() -> {
+            AutoFollowStats autoFollowStats = getAutoFollowStats();
+            assertThat(autoFollowStats.getNumberOfSuccessfulFollowIndices(), equalTo(1L));
+            assertThat(autoFollowStats.getNumberOfFailedFollowIndices(), equalTo(0L));
+            assertThat(autoFollowStats.getNumberOfFailedRemoteClusterStateRequests(), equalTo(0L));
+        });
+        assertTrue(ESIntegTestCase.indexExists("copy-logs-201701", followerClient()));
+
+        createLeaderIndex("logs-201801", leaderIndexSettings);
```

Comment:
The `assertLongBusy` below would never wait (unless it fails). Like in my other comment, I think the excluded index should be created before the non-excluded one so that we can verify it is not followed when the non-excluded index has been followed.

Reply:
🤦 sorry, I missed this one. I've addressed it in https://github.com/elastic/elasticsearch/pull/72935/commits/c531a609d17685f7fd977cfbde6280edc19edb1a